test/onnx/test_pytorch_onnx_shape_inference.py

```diff
@@ -268,6 +271,60 @@ def test_resize_after_concat(self):
         )
         self.run_test(g, resize.node(), expect_tensor("Float", shape=(4, 32, 128, 128)))
 
+    def test_autograd_layernorm_shape(self):
+        def symbolic_pythonop(ctx: torch.onnx.SymbolicContext, g, *args, **kwargs):
+            return g.op("com.microsoft::PythonOp")
+
+        torch.onnx.register_custom_op_symbolic("prim::PythonOp", symbolic_pythonop, 1)
+        self.addCleanup(torch.onnx.unregister_custom_op_symbolic, "prim::PythonOp", 1)
+
+        # necessay parameters for transformer embeddings
+        hidden_size, max_position_embeddings, batch_size = (
+            48,
+            32,
+            2,
+        )
+
+        # wrap nn.layernorm into autograd.func repro the same
+        # issue as using apex fusedlayernorm
+        class CustomLayerNorm(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, embedding):
+                layer_norm = torch.nn.LayerNorm(hidden_size, eps=1e-12)
+                return layer_norm(embedding)
+
+        class EmbeddingModule(torch.nn.Module):
+            def forward(
+                self,
+                embeddings=None,
+            ):
+                embedding_output = CustomLayerNorm.apply(embeddings)
+                query = embedding_output.transpose(0, 1)
+                tgt_len, bsz, embed_dim = query.size()
+                query = query.reshape(tgt_len, bsz, embed_dim)
```

Comment:
Are we reshaping query to the exact same shape it used to have? If the step is needed for the repo I would add a comment explaining why it is needed

Reply:
I added the explanation for this reshape usage. It consumed the batch_size to expose the fact that it becomes static.